### PR TITLE
Run the tests using GitHub CI

### LINKED ISSUE #25: Migrate the CI service
[travis-ci._**org**_](https://www.travis-ci.org/) is shutting down soon, and has stopped doing builds for `mc706/changelog-cli`.

We have to migrate to a different system.

While we could migrate to the new [travis-ci._**com**_](https://www.travis-ci.com) (which is configured using GitHub Actions), I would prefer to migrate to something else:

In order of preference
1. GitHub Actions itself
2. CircleCI
3. TravisCI

I have experience with GitHub Actions and CircleCI for other projects.

* [Here is an example](https://github.com/movermeyer/backports.datetime_fromisoformat/blob/master/.github/workflows/test.yml) of a simple build I use GitHub Actions for.
* [Here is an example](https://github.com/closeio/ciso8601/blob/master/.circleci/config.yml) of a more complicated project that I use CircleCI for. I would use GitHub Actions for this too, but it will soon be a much more complicated build (once I add binary wheels; but that's another story)

### PR DESCRIPTION
**Pull request by movermeyer**
Fixes #25

Runs the tests, mypy, pyre, and prospector on every commit.

As compared to the existing [TravisCI config](https://github.com/mc706/changelog-cli/blob/cb94acca75f63bab07875c9c9d2d726c63d09b1c/.travis.yml):

- This doesn't test against `nightly`. It doesn't appear that GitHub Actions offers that.
- This runs the static analysis tools

### CODE REVIEW

**Comment by mc706** [comment]
Im taking a look into seeing if we can use the `matrix.version` flag as a pass-in variable for the contianer. That way we could use images like `python-3.7:alpine` which would allow us to not reinstall python on every run and download _much_ faster. 

However that is also thinking coming from a paid gitlab account where we paid for execution time instead of having free actions because it is open source. So less of a benefit but I still feel like `faster_ci > slower_ci` nevertheless.

**Comment by movermeyer** [comment]
> I'm taking a look into seeing if we can use the matrix.version flag as a pass-in variable for the container.

@mc706 

GitHub Actions unfortunately doesn't work like that. You can't choose an arbitrary Docker image as the base image.
You use one of their "Runners" using [`runs-on`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on) that forms the base machine.

That said, the `ubuntu-latest` runner [already has the latest version of each of the versions of Python](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#pypy) that we are using, so no time is lost installing Python itself.

File: .coveragerc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [run]
 branch = True
+relative_files = True
```
Comment (movermeyer) [inline .coveragerc:3]: Needed for https://github.com/AndreMiras/coveralls-python-action

File: src/changelog/utils.py [hunk 2]
```diff
@@ -81,10 +81,10 @@ def get_unreleased_change_types(self) -> Set[str]:
                 if line in self.REVERSE_SECTIONS:
                     section = self.REVERSE_SECTIONS[line]
                     continue
-                elif line in self.BETA_REVERSE_SECTIONS:
+                if line in self.BETA_REVERSE_SECTIONS:
                     section = self.BETA_REVERSE_SECTIONS[line]
                     continue
-                change_types.add(section)
+                change_types.add(cast(str, section))
```
Comment (movermeyer) [inline src/changelog/utils.py:87]: Fix for mypy/pyre

File: src/changelog/utils.py [hunk 3]
```diff
@@ -81,10 +81,10 @@ def get_unreleased_change_types(self) -> Set[str]:
                 if line in self.REVERSE_SECTIONS:
                     section = self.REVERSE_SECTIONS[line]
                     continue
-                elif line in self.BETA_REVERSE_SECTIONS:
+                if line in self.BETA_REVERSE_SECTIONS:
```
Comment (movermeyer) [inline src/changelog/utils.py:84]: Fix for prospector

**Review by mc706 — APPROVED** [review]

**Pull request merged**